### ISSUE TITLE
All nodes failed after reset for new agent!

### ISSUE THREAD

**Issue by DaHoopster** [Shippable/support#3486]
### Description of your issue:
Got a notice on the shippable dashboard on a new agent. I was asked to reset the nodes, which I did. But after resetting, I received failure messages on all nodes. Our builds are not working right now.

See screenshot:
<img width="1372" alt="screen shot 2017-03-31 at 1 22 27 pm" src="https://cloud.githubusercontent.com/assets/2730866/24548383/219305f0-1615-11e7-90a6-54b241050aa0.png">

- Link to the build failure
- Scripts called from the yml (optional)

**Comment by DaHoopster** [comment]
Oh, things are good now

**Comment by himanshu0503** [comment]
@DaHoopster : We see that your nodes have an error log
```
Warning: failed to get default registry endpoint from daemon (Cannot connect to the Docker daemon. Is the docker daemon running on this host?). Using system default: https://index.docker.io/v1/
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```
Reopening this to investigate.

**Comment by himanshu0503** [comment]
@DaHoopster : Can you please provide us the following info from your machines? 
1. Output of `docker ps`
2. Output of `docker images`
Please mail me at himanshu@shippable.com if you don't want this information to be public.

**Comment by DaHoopster** [comment]
OK, thanks @himanshu0503. Do you need the output for all 4 machines?

**Comment by DaHoopster** [comment]
On one of the machines:

```
$docker ps
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

```
$docker images
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Comment by himanshu0503** [comment]
@DaHoopster : Can you try doing it with sudo?

**Comment by DaHoopster** [comment]
Ahh yes

```
~$ sudo docker ps

CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS               NAMES
ff177b4f02eb        shipimg/genexec:v5.3.4   "/home/shippable/micr"   27 hours ago        Up 23 minutes                           shippable-exec-58dcc3cb30240a0700768338
```

```
~$ sudo docker images

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
shipimg/genexec     v5.3.4              91e3939f36fc        7 days ago          2.393 GB
drydock/u14ruball   prod                4779a48166ba        6 months ago        7.11 GB
```

**Comment by chetantarale** [comment]
@DaHoopster Looks like its pulling the old image which is not expected. Can we jump on a quick hangout.
https://hangouts.google.com/hangouts/_/shippable.com/support?hl=en&authuser=0

**Comment by himanshu0503** [comment]
@DaHoopster : As per the logs, old shippable agent is still running on your system. Do you mind giving the output of the following commands on all the machines
```
1. sudo docker ps -a
2. sudo docker images
3. sudo df -h
4. sudo free -m
```

**Comment by DaHoopster** [comment]
```
$ sudo docker ps -a

CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS               NAMES
ff177b4f02eb        shipimg/genexec:v5.3.4   "/home/shippable/micr"   29 hours ago        Up About an hour                        shippable-exec-58dcc3cb30240a0700768338
```

```
$ sudo docker images

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
shipimg/genexec     v5.3.4              91e3939f36fc        7 days ago          2.393 GB
drydock/u14nodall   prod                2c5bea464365        6 months ago        5.144 GB
drydock/u14ruball   prod                4779a48166ba        6 months ago        7.11 GB
```

```
$ sudo df -h

Filesystem      Size  Used Avail Use% Mounted on
udev            2.0G  8.0K  2.0G   1% /dev
tmpfs           396M  448K  395M   1% /run
/dev/xvda1       30G   15G   14G  52% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
none            5.0M     0  5.0M   0% /run/lock
none            2.0G  264K  2.0G   1% /run/shm
none            100M     0  100M   0% /run/user
none             30G   15G   14G  52% /data/aufs/mnt/e22a4ba9274f1508e60b1314ee9688ce2ab0dcf521e6c21fc40906eb737b355f
shm              64M     0   64M   0% /data/containers/ff177b4f02eb767c58962746d362325903c2c5318d51e552af73503abbd7f76f/shm
```

```
$ sudo free -m

             total       used       free     shared    buffers     cached
Mem:          3951       1812       2138          0        227       1031
-/+ buffers/cache:        553       3397
Swap:            0          0          0
```

**Comment by DaHoopster** [comment]
@himanshu0503 any updates?

**Comment by himanshu0503** [comment]
@DaHoopster : We are looking into it but haven't been able to figure out a repro step at our end. On two of machines which failed with memory error but it looks like there is enough memory available. Can we get on a hangout to figure this out?

**Comment by lucasmpb** [comment]
Hello, im having the same issue. After reset the nodes to upgrade them, they failed and show this in the console
```
Warning: failed to get default registry endpoint from daemon (Cannot connect to the Docker daemon. Is the docker daemon running on this host?). Using system default: https://index.docker.io/v1/
Cannot connect to the Docker daemon. Is the docker daemon running on this host?


Node  init script completed
```
And
```
Command failed : /tmp/67adc4a9-675e-4add-9b0c-e66c680d06c2/67adc4a9-675e-4add-9b0c-e66c680d06c2-0.sh 67adc4a9-675e-4add-9b0c-e66c680d06c2 0 Exception Script failure tag received
```

**Comment by trriplejay** [comment]
one issue i'm seeing in some of the logs is docker failing to restart during the initialization:
```
# 19:41:29 #######################################
# Starting restart_docker_service
##################################################
checking if docker restart is necessary
restarting docker service on reset
|___ Running sudo service docker restart
docker stop/waiting
start: Job failed to start
```

it might be helpful to check the docker logs on the machine:

`sudo tail -n50 /var/log/upstart/docker.log`

it might give some clue as to the cause of the errors.

**Comment by DaHoopster** [comment]
@himanshu0503 we have to wait on the hangout until next week. I am currently traveling for work in Paris. Can we schedule a hangout next Tuesday? I will be back in San Francisco.

**Comment by chetantarale** [comment]
@DaHoopster we have figured out the problem.
The problem here is your nodes were initialized with our old scripts and when you reset(for the first time) them they are using the new scripts. The new scripts are appending `DOCKER_OPTS` with the dns. This is duplicating the `DOCKER_OPTS` entry and hence docker daemon is failing to respond.

We have filed an internal issue to fix this. meanwhile resetting the nodes should work. 
When you reset it should be pulling the latest `genexec:v5.4.1` image.
